### PR TITLE
Add const lvalue ref to `core/*` container parameters

### PR DESCRIPTION
**Pull request by Muller-Castro**
Adds const lvalue reference to the core container types specified in the [documentation](https://docs.godotengine.org/en/stable/development/cpp/core_types.html) that are declared in function parameters in `core/*` files

initial: https://github.com/godotengine/godot/pull/51156
`editor/*`: https://github.com/godotengine/godot/pull/88368
`modules/*`: https://github.com/godotengine/godot/pull/88915
`platform/*`: https://github.com/godotengine/godot/pull/88971
`servers/*`: https://github.com/godotengine/godot/pull/88972
`scene/*`: https://github.com/godotengine/godot/pull/88974
others: https://github.com/godotengine/godot/pull/88975

### CODE REVIEW

**Comment by AThousandShips** [comment]
This doesn't just cover core, you need to split it down further place, let's keep it to just `core/`

File: core/core_bind.cpp
Comment (AThousandShips) [inline core/core_bind.cpp:55]: Not valid, this should be:
```suggestion
ResourceLoader::ThreadLoadStatus ResourceLoader::load_threaded_get_status(const String &p_path, Array &r_progress) {
```
If changed at all

File: core/core_bind.cpp
Comment (AThousandShips) [inline core/core_bind.cpp:284]:
```suggestion
int OS::execute(const String &p_path, const Vector<String> &p_arguments, Array &r_output, bool p_read_stderr, bool p_open_console) {
```

**Comment by AThousandShips** [comment]
Please limit it to **just** core, this is still too big to do a proper review of for these kinds of changes 🙂 no harm in multiple PRs, especially when it makes it possible to review

**Comment by Muller-Castro** [comment]
I tried to just upload the `core/*` files but it was issuing build errors because of `core/*` virtual methods overridden across multiple parts of the engine

**Comment by AThousandShips** [comment]
Then those should be handled separately in their own PR because this still has over 300 files, far too big for a review like this, for those specific methods you can do a dedicated PR

**Comment by Muller-Castro** [comment]
Ok I'll leave the virtual methods that are issuing build errors out of this PR and put then in another PR

**Comment by AThousandShips** [comment]
Or at minimum remove all unrelated changes from other parts, because they make up a lot of it 🙂

**Comment by Muller-Castro** [comment]
You mean remove all folders except `core` and core's virtual methods that aren't compiling?

File: core/io/dir_access.cpp
Comment (AThousandShips) [inline core/io/dir_access.cpp:463]: For now I'd remove any case that requires copying in the body, let's keep this to just const changes, and possibly changing the iterators if needed

**Review by AThousandShips — COMMENTED** [review]
Initial glance, will go over properly when the scope has been reduced 🙂

File: core/io/dir_access.cpp
Comment (Muller-Castro) [reply core/io/dir_access.cpp:463]: Got it!

**Comment by Muller-Castro** [comment]
Those non-`core/*` are inevitable for all `core/*` files to build

**Comment by AThousandShips** [comment]
Will take a look, thank you and sorry for the inconvenience (and you'll get a bonus of more PRs!)

File: core/variant/method_ptrcall.h
Comment (AThousandShips) [inline core/variant/method_ptrcall.h:1]: I need a second eye on this, as it's very core and specific, unsure about possible side effects, they're my only question mark as it were, should be fine but it's so central that possible side effects could be major

**Review by AThousandShips — APPROVED** [review]
Other than the note below this looks good and safe to me

**Comment by AThousandShips** [comment]
You have a merge commit in your history, please use rebasing instead, do this by:
```
git reset --hard HEAD^
git rebase -i master
git push --force
```

**Comment by akien-mga** [comment]
Sorry for the delay. Looks good to me overall, needs a rebase and then this should be ready to merge if nobody has any outstanding concerns about the change.

**Comment by AThousandShips** [comment]
I'd like an eye by a core maintainer familiar with the `method_ptrcall.h` file, just to ensure there's no weirdness with that, otherwise I think this looks good

CC @reduz perhaps

**Comment by dsnopek** [comment]
The changes in `method_ptrcall.h` seem fine to me! The `const` is added to `Vector`'s that we're only reading from.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
